appsec-kit-starter/src/test/java/com/vaadin/appsec/service/AppSecDevToolsPluginTest.java

```diff
@@ -0,0 +1,127 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class AppSecDevToolsPluginTest {
+
+    private AppSecDevToolsPlugin appSecDevToolsPlugin;
+    private DevToolsInterface devToolsInterface;
+    private MockedStatic<AppSecService> appSecService;
+    private AppSecService appSecServiceInstance;
+    private Registration registration;
+
+    @Captor
+    private ArgumentCaptor<AppSecScanEventListener> appSecScanEventListenerCaptor;
+    @Captor
+    private ArgumentCaptor<String> commandCaptor;
+    @Captor
+    private ArgumentCaptor<JsonObject> jsonObjectCaptor;
+
+    @BeforeEach
+    public void setup() {
+        appSecDevToolsPlugin = new AppSecDevToolsPlugin();
+        devToolsInterface = mock(DevToolsInterface.class);
+
+        appSecServiceInstance = mock(AppSecService.class);
+        registration = mock(Registration.class);
+        when(appSecServiceInstance.addScanEventListener(any()))
+                .thenReturn(registration);
+        var vulnerabilities = Collections
+                .singletonList(new Vulnerability("CVE-000"));
+        when(appSecServiceInstance.getNewVulnerabilities())
+                .thenReturn(vulnerabilities);
+
+        appSecService = mockStatic(AppSecService.class);
+        appSecService.when(AppSecService::getInstance)
+                .thenReturn(appSecServiceInstance);
+    }
+
+    @AfterEach
+    public void cleanup() {
+        appSecService.close();
+    }
+
+    @Test
+    public void handleConnect_sendsCommands() {
+        var vulnerabilities = Arrays.asList(new Vulnerability("CVE-001"),
+                new Vulnerability("CVE-002"));
+        var appSecScanEvent = mock(AppSecScanEvent.class);
+        when(appSecScanEvent.getNewVulnerabilities())
+                .thenReturn(vulnerabilities);
+
+        appSecDevToolsPlugin.handleConnect(devToolsInterface);
+
+        verify(appSecServiceInstance)
+                .addScanEventListener(appSecScanEventListenerCaptor.capture());
+        var appSecScanEventListener = appSecScanEventListenerCaptor.getValue();
+        appSecScanEventListener.scanCompleted(appSecScanEvent);
+
+        verify(devToolsInterface, times(3)).send(commandCaptor.capture(),
+                jsonObjectCaptor.capture());
+        var commands = commandCaptor.getAllValues();
+        assertEquals("appsec-kit-init", commands.get(0));
+        assertEquals("appsec-kit-scan", commands.get(1));
+        assertEquals("appsec-kit-scan", commands.get(2));
+        var data = jsonObjectCaptor.getAllValues();
+        assertNotNull(data.get(0));
+        assertEquals(1, data.get(1).get("vulnerabilityCount").asNumber());
+        assertEquals(2, data.get(2).get("vulnerabilityCount").asNumber());
```

Comment:
To further improve the likelihood of the chronological sequence of events, what if:
1. we first verify that `appsec-kit-init` and the first `appsec-kit-scan` messages are sent, i.e. _before_ calling `scanCompleted` (and also verify the vulnerability count of that first message)
2. then we call `scanCompleted` and we verify we get the second `appsec-kit-scan` message with the new vulnerability count

Reply:
Good idea. Done.